wandb/sdk/artifacts/artifact.py

```diff
@@ -679,24 +676,24 @@ def file_count(self) -> int:
 
     @property
     def created_at(self) -> str:
-        """The time at which the artifact was created."""
+        """Timestamp when the artifact was created."""
         self._ensure_logged("created_at")
         assert self._created_at is not None
         return self._created_at
 
     @property
     def updated_at(self) -> str:
-        """The time at which the artifact was last updated."""
+        """The time when the artifact was last updated."""
         self._ensure_logged("updated_at")
         assert self._created_at is not None
         return self._updated_at or self._created_at
 
     # State management.
 
     def finalize(self) -> None:
-        """Mark this artifact as final, disallowing further modifications.
+        """Finalize the artifact. You can not modify an artifact once it is finalized.
```

Comment:
```suggestion
        """Finalize the artifact. You can not modify an artifact once it is finalized.
```
@ibindlish what does it mean to finalize an artifact? It prevents it from having any new versions logged to it? Is this mostly for distributed artifacts where several runs are logging to it?

Reply:
Answered below

Reply:
Updated description. How does that look?

Reply:
almost perfect - lets just change "you **cannot** modify an artifact **version**..." (add version and correct can not)